crates/sage/src/scoring.rs

```diff
@@ -385,18 +400,20 @@ impl<'db> Scorer<'db> {
         let mz = precursor.mz - PROTON;
 
         for idx in 0..report_psms.min(score_vector.len()) {
-            let score = score_vector[idx];
+
+            let score = score_vector[idx].0;
+            let fragments = score_vector[idx].1.clone();
```

Comment:
Cloning is unacceptable here, too expensive. 

Along with my comment about only generating annotations as needed, one way to do it could be to do it in a separate pass from scoring - take the MS2 spectra and the peptide, and re-generate the fragment ion series again and annotate.

Reply:
I believe that the std::mem::swap function is a suitable choice for swapping the contents of two values in a workplace setting. This function is specifically designed for this purpose and takes two mutable references as input. It efficiently swaps the values without the need to copy the actual data. The cost of swapping is generally minimal, as it involves swapping pointers and sizes instead of moving the entire content.

On the other hand, std::mem::replace is a function used to replace the value in a mutable reference with a new value, while also returning the old value. However, it involves moving the entire value out and replacing it with a new one. The cost of using std::mem::replace can be higher, especially if the value being replaced is large or expensive to move.

Reply:
Both mem::swap and replace work - but I settled for `Option::take` which is the most convenient function to use here